File: 2018/puzzle2/js/solution.js
const fs = require('fs')

fs.readFile('../input.txt', 'utf8', (err, data) => {
    let numbers = data.split('\n').map(numStr => Number(numStr))
    numbers.pop()
    let currFreq = 0
    let idx = 0
    let pastFreqs = new Set()

    while (true) {
        if (idx === numbers.length) idx = 0;
        if (pastFreqs.has(currFreq)) break;
        pastFreqs.add(currFreq)
        currFreq += numbers[idx]
        idx++
    }
    
    console.log(`The first repeated frequency is ${currFreq}`)
})

File: 2018/python/puzzle2/solution.py
"""Advent of Code, Puzzle 2 solution."""
if __name__ == "__main__":
    with open('input.txt') as data:
        numbers = [int(number) for number in data.readlines()]
        current_freq = 0
        idx = 0
        past_frequencies = set()
        while True:
            if idx == len(numbers):
                idx = 0
            if current_freq in past_frequencies:
                break
            past_frequencies.add(current_freq)
            current_freq += numbers[idx]
            idx += 1
        print(f"The first repeated frequency is {current_freq}")
